### PR TITLE
Mafia can be played on your PDA

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Mafia is now friggin playable from the PDA, I also changed other stuff too

- You can't use abilities on dead people if you're not supposed to (cant kill the same person over and over)
- Changelings cant kill other Changelings
- Changelings can now only talk to eachother at night, rather than using :j
- Everyone starts spawned in the center of the map, since people playing on PDA can't move their characters. This is so everyone can hear PDA users in person, as I don't want the chat log to be mandatory.

To do this, all messages you are meant to be able to see, is now logged for you to see in your Mafia panel. This essentially means that people playing through the PDA get a downgraded version of it, but I don't know how much larger I want this UI to be.

Playing Mafia through the PDA will not tell you of other players ahead of time when signing up (as it shows ckeys + pdas), but they can see the names in-game. Unfortunately this means we'll have to remove your customization coming with you, to prevent using it to tell who is dead in round.

Things I am missing
- Program overlays on PDA/Laptop/Computer
- Icon for the app's header while a game is active

I'm not a spriter and can't make either of these

This is the new UI
image

I also fixed alert calls for PDAs and stuff
image

and removed the X at the top on computers since they had no battery
image

Looks a little better now hopefully 👍 

## Why It's Good For The Game

- The current Arcade app sucks, and is a solo game. This is much more entertaining and you can talk to others in it, which is swag as fuck.
- There's a larger potential playerbase for the Minigame making it more likely to be played.
- Sets groundwork for a better version of https://github.com/tgstation/tgstation/pull/75879
- Adds more suspense and teamwork in the minigame.

## Changelog

:cl: JohnFulpWillard, sprites by CoiledLamb
add: You can now play Mafia on your PDA.
balance: Mafia changelings can now only talk to eachother during the night.
fix: Mafia abilities can't be repeatedly used on people.
/:cl:

### CODE REVIEW

**Comment by bawbbawbins** [comment]
jhony wanna paly bloxburg later today

**Comment by JohnFulpWillard** [comment]
i dont have a roblox account

**Comment by Y0SH1M4S73R** [comment]
> Unfortunately this means we'll have to remove your customization coming with you, to prevent using it to tell who is dead in round.

Would it be feasible to give ghost players a pseudonym that appears when they are referenced on the PDA, or would it be too difficult to catch instances where part of a character's name is used in speech?

**Comment by JohnFulpWillard** [comment]
> Would it be feasible to give ghost players a pseudonym that appears when they are referenced on the PDA, or would it be too difficult to catch instances where part of a character's name is used in speech?

I think it would ruin part of the experience, calling someone by their first/last/short names would not be caught and then you'd have confusion of 2 groups of people calling someone by 2 different names.

**Comment by Dawnseer** [comment]
Alright but what if I got someone else's PDA, and I use both my own PDA and theirs to play. Will I be able to rig the game?

**Comment by JohnFulpWillard** [comment]
> Alright but what if I got someone else's PDA, and I use both my own PDA and theirs to play. Will I be able to rig the game?

Yea

**Comment by Dawnseer** [comment]
Sweet. I am now imagining the HoP and some morgue looter playing a full game of mafia but it's just those two playing all the roles.

**Comment by JohnFulpWillard** [comment]
Names are all anonymous so it'll be harder to figure out who is actually playing, outside of the alert that a new game started (which I could remove)- and that's just life I suppose.

**Comment by Dawnseer** [comment]
Just gotta normalize centcomm sending deathsquads after people who cheat at mafia. Problem solved, and fun for all parties involved. Probably.

**Comment by EricZilla** [comment]
I cannot beleive we can metacomm IC now

**Comment by JohnFulpWillard** [comment]
preferably you would not

**Comment by JohnFulpWillard** [comment]
I'm gonna have to close this until I got sprites for it

**Comment by Time-Green** [comment]
😢

File: code/modules/mafia/controller.dm [hunk 1]
```diff
@@ -584,21 +608,28 @@ GLOBAL_LIST_INIT(mafia_role_by_alignment, setup_mafia_role_by_alignment())
...
 		role.greet()
 
-/datum/mafia_controller/ui_static_data(mob/user)
+/datum/mafia_controller/ui_static_data(atom/user)
 	var/list/data = list()
 
-	if(user.client?.holder)
+	if(usr.client?.holder)
 		data["admin_controls"] = TRUE //show admin buttons to start/setup/stop
+	data["is_observer"] = !!isobserver(user)
```
Comment (Ghommie) [inline code/modules/mafia/controller.dm:627]: `istype()` already return booleans.

File: code/modules/mafia/controller.dm [hunk 2]
```diff
@@ -619,18 +650,25 @@ GLOBAL_LIST_INIT(mafia_role_by_alignment, setup_mafia_role_by_alignment())
...
 
-	data["timeleft"] = next_phase_timer ? timeleft(next_phase_timer) : 0 //the tgui menu counts this down.
+	data["timeleft"] = next_phase_timer ? timeleft(next_phase_timer) : 0
 
 	var/datum/mafia_role/user_role = player_role_lookup[user]
 	if(user_role)
 		data["user_notes"] = user_role.written_notes
+		data["messages"] = list()
+		var/list/ui_messages = list()
+		for(var/i = user_role.role_messages.len to 1 step -1)
```
Comment (Ghommie) [inline code/modules/mafia/controller.dm:667]: A part of me wants to suggest `reverseList`, but it's no big deal really.

**Review by Ghommie — CHANGES_REQUESTED** [review]
You need to make the `award_role` proc compatible with modular computer / PDA players, if you haven't, otherwise lgtm except one and a half nitpicks.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**